Sources/ImageOptions.swift

```diff
@@ -0,0 +1,335 @@
+//
+//  Image.swift
+//  Contentful
+//
+//  Created by JP Wright on 24.05.17.
+//  Copyright © 2017 Contentful GmbH. All rights reserved.
+//
+
+import Foundation
+import CoreGraphics
+
+extension Asset {
+
+    /**
+     The URL for the underlying media file with additional options for server side manipulations
+     such as format changes, resizing, cropping, and focusing on different areas including on faces,
+     among others.
+
+     - Parameter imageOptions: An array of `ImageOption` that will be used for server side manipulations.
+     - Throws: Will throw SDKError if the SDK is unable to generate a valid URL with the desired ImageOptions.
+     */
+    public func url(with imageOptions: [ImageOption] = []) throws -> URL {
+
+        // Check that there are no two image options that specifiy the same query parameter.
+        // https://stackoverflow.com/a/27624476/4068264z
+        // A Set is a collection of unique elements, so constructing them will invoke the Equatable implementation
+        // and unique'ify the elements in the array.
+        let uniqueImageOptions = Array(Set<ImageOption>(imageOptions))
+        guard uniqueImageOptions.count == imageOptions.count else {
+            throw SDKError.invalidImageParameters("Cannot specify two instances of ImageOption of the same case."
+                + "i.e. `[.formatAs(.png), .formatAs(.jpg(withQuality: .unspecified)]` is invalid.")
+        }
+        guard imageOptions.count > 0 else {
+            return try url()
+        }
+
+        let urlString = try url().absoluteString
+        guard var urlComponents = URLComponents(string: urlString) else {
+            throw SDKError.invalidURL(string: urlString)
+        }
+
+        urlComponents.queryItems = try imageOptions.flatMap { option in
+            try option.urlQueryItems()
+        }
+
+        guard let url = urlComponents.url else {
+            throw SDKError.invalidURL(string: urlString)
+        }
+        return url
+    }
+}
+
+
+public enum ImageOption: Equatable, Hashable {
+
+    /// Specify the size of the image in pixels to be returned from the API. Valid ranges for width and height are between 0 and 4000.
+    case sizedTo(width: Int, height: Int)
```

Comment:
Can I specify a `width` only? Aka, I want to scale the image to 100 pixel width, and let the aspect ratio kick in to calculate the `height`?